### PR TITLE
Sprint planning content v1

### LINKED ISSUE #56: content for planning
This is for the new proposed outline in #53 

Using the suggested template below as a guide, describe the concept of planning in scrum. This should include (but not limited to)

- Inputs: What are the pre-reqs for planning (i.e. estimated & prioritized stories)
- Participants
- Outputs: What decisions or actions are needed to conclude the meeting?
- Determining team capacity
- Tasking stories in sprint backlog
     - Size of tasks
     - Hour estimates
     - Updating todo hours
- Dividing work into WIP (work-in-progress) or Workstreams

### Goals
What are the intended outcomes of the ceremonies?

### Participation
What members of the team should participate?

### Impact
What positive impact has been observed as a result of this practice?

### Measures
How might one measure the impact of this ceremony positive or negative?

### Facilitation Guidance
How might one go about running the ceremony? Is there a general pattern that the ceremony should follow?

### PR DESCRIPTION
**Pull request by meulta**
First iteration for the description of the sprint planning.

Fixes #56

### CODE REVIEW

**Comment by meulta** [comment]
Fixed the markdown issues :)

File: sprint-planning/readme.md [hunk 1]
```diff
@@ -1,29 +1,83 @@
...
+**Goals are:**
+
+- Selecting the **stories** that will be implemented in the sprint.
+- Estimating the **effort** required for the stories in the sprint.
+- Splitting the stories into **tasks**.
+
+General guidance:
 
-What are the intended outcomes of the ceremonies?
+- The sprint planning should happen at the begining of the new sprint (or at the end of the previous one). It usually last bewteen 1 and 4 hours depending on the size of the team and duration of the sprints.
```
Comment (fnocera) [inline sprint-planning/readme.md:15]: I would question planning lasting longer than 2 hours....would be interested in hearing what others think.

File: sprint-planning/readme.md [hunk 2]
```diff
@@ -1,29 +1,83 @@
...
+- Each story should be achievable in the sprint. Otherwise, you should split the story.
```
Comment (fnocera) [inline sprint-planning/readme.md:16]: Nit: "Each story should be able to be completed within the length of the sprint. Otherwise, the story should be broken up into multiple stories."

File: sprint-planning/readme.md [hunk 3]
```diff
@@ -1,29 +1,83 @@
...
+  
+    - Describes to the team the stories that they propose for the sprint.
+
+- **All team members**:
+
+  - Listen to the product owner story description.
+  - Ask questions to make sure everyone understands each story properly.
+  - [Estimate](./estimation/readme.md) the effort for each backlog item, as a team.
+  - Split each story into tasks.
+  - (Optional) self assign first task to team members.
```
Comment (fnocera) [inline sprint-planning/readme.md:46]: I am surprised that in planning you would have time to split stories into tasks, usually this is left to whoever owns the user story. 

I have seen stories be assigned to team members at planning vs tasks.

File: sprint-planning/readme.md [hunk 4]
```diff
@@ -1,29 +1,83 @@
...
+
+*(\*) some teams find useful to define a **[Definition of ready](https://www.agilealliance.org/glossary/definition-of-ready)** that describes the list of things that needs to be done in each story before the **product owner** can propose it for a **print**. The list proposed here is the classic minimal definition of ready.*
```
Comment (fnocera) [inline sprint-planning/readme.md:48]: Would reword to something like: "It can be helpful to determine a "Definition of ready". This describes what needs to be done for a story to be "ready" to be introduced into planning." 

Not 100% sure what this means "before the **product owner** can propose it for a **print**", would remove.

File: sprint-planning/readme.md [hunk 5]
```diff
@@ -1,29 +1,83 @@
...
+**Sprint planning key benefits:**
+
+- Everyone participates, the entire team is aware of the scope of the sprint.
+- Each team member takes responsibility in the sprint scope by participating in the stories discovery, sign up, prioritization and estimation.
+- Creates a platform to communicate, discover and discuss dependencies.
 
 ## Measures
 
-How might one measure the impact of this ceremony positive or negative?
+- How many stories needed more work before being presented (and were rejected because not clear enough)?
```
Comment (fnocera) [inline sprint-planning/readme.md:60]: Nit: "How many stories needed more work before being presented (and were rejected because they were not clear enough)?"

File: sprint-planning/readme.md [hunk 6]
```diff
@@ -1,29 +1,83 @@
...
+- Confirming team capacity. This should be done using the average velocity (number of points achieved per sprint) and removing the project vacations/holidays/off days. If it helps, you can get the daily average velocity, divide by the number of team members and multiply by number of off days.
+- Timebox (for instance: half of the meeting for story selection, half of the meeting for task splitting).
+- Agree on how much capacity needs to be "saved" for bug fixing (might depend on the sprint).
+- Ensure everyone understands each story that is selected for the sprint.
+- Ensure everyone participates in story effort estimation.
+
+**Other considerations:**
+
+- Take into account off days (vacations, national holidays, unavailability).
+- If the amount of work in a Sprint is is too much to handle, you might want to split into different workstreams. This might require thinking about [scrum of scrums](./../scrum-of-scrums/readme.md) and all related ceremonies.
```
Comment (fnocera) [inline sprint-planning/readme.md:83]: Nit: "too much to handle" this reads a little informal/colloquial, would re-word.

File: sprint-planning/readme.md (hunk 1 above)
Comment (bsherwin) [reply sprint-planning/readme.md:15]: I think the "depending on the size of the team and duration of the sprints"...if you have 4 week sprints (not CSE) you might need four hours of planning...one week sprints not so much.

File: sprint-planning/readme.md (hunk 3 above)
Comment (bsherwin) [reply sprint-planning/readme.md:46]: First, I would say the team owns the story...the team owns the code...the team owns the sprint. There is no story owner...maybe a lead is assigned, but collectively we own the success of the sprint.

In planning, if there is a high priority story, it may be that the team would need to work together to get it done as it may be a blocker for other stories prioritized for that sprint. Taking some time to task out the story can find opportunities for parallel or dependent work. But you wouldn't necessarily need to task out every story (especially small ones).

File: sprint-planning/readme.md (hunk 4 above)
Comment (bsherwin) [reply sprint-planning/readme.md:48]: We have a "definition of ready" doc somewhere in this playbook.

**Comment by debugerr** [comment]
Forgot to submit my review comments that I made a few days ago. Unfortunately, I'm out for the next 2 weeks, hence I just submitted them as comments (neither waiting for author nor approving). Treat them as nice to have or food for thoughts.

File: sprint-planning/readme.md (hunk 1 above)
Comment (meulta) [reply sprint-planning/readme.md:15]: I am happy to reduce this. 4 hours at needed in 4 weeks sprints. If this only target what we are doing in CSE today (1 week sprints), then it makes more sense to on indicate 1-2 hours

File: sprint-planning/readme.md (hunk 2 above)
Comment (meulta) [reply sprint-planning/readme.md:16]: Fixed in upcoming PR update

File: sprint-planning/readme.md (hunk 3 above)
Comment (meulta) [reply sprint-planning/readme.md:46]: Agreed with Brian. The classic recommendation is to divide the sprint planning in 2 parts:

- One during which the team build the backlog at the story level
- One during which the team splits stories into tasks, when it make sense.

This ensures the entire team is aware of everything that needs to be done.

This is the typical recommendation. But I am happy to add something here that makes it optional.

File: sprint-planning/readme.md (hunk 4 above)
Comment (meulta) [reply sprint-planning/readme.md:48]: Will add the link. Also, @fnocera : it was missing an S. "before the product owner can propose it for a Sprint" :) 

Fixed in an upcoming PR update.

File: sprint-planning/readme.md (hunk 6 above)
Comment (meulta) [reply sprint-planning/readme.md:83]: Proposed change in upcoming PR update.
Keeping comment thread opened for now

File: sprint-planning/readme.md [hunk 7]
```diff
@@ -1,29 +1,72 @@
...
 
-## Goals
+**Goals are:**
+- Selecting the **stories** that will be implemented in the sprint.
+- Estimating the **effort** required for the stories in the sprint.
+- Splitting the stories into **tasks**.
 
-What are the intended outcomes of the ceremonies?
+General guidance:
+- The sprint planning should happen at the begining of the new sprint (or at the end of the previous one). It usually last bewteen 1 and 4 hours depending on the size of the team and duration of the sprints.
```
[the remark this answers is not in the document]
Comment (meulta) [reply sprint-planning/readme.md:15]: Thanks. Fix in upcoming PR update

File: sprint-planning/readme.md [hunk 8]
```diff
@@ -1,29 +1,72 @@
...
+     - Is in the correct position in the backlog, by right priority order.
+     - Contains a clear description.
+     - Contains a clear [definition of done](https://www.agilealliance.org/glossary/definition-of-done/).
+     - Do NOT pre assigns stories to the future sprint. This is the purpose of the sprint planning.
+  - During the meeting: 
+    - Describes to the team the stories that they propose for the sprint.
+-  **All team members**:
+   - Listen to the product owner story description.
+   - Ask questions to make sure everyone understands each story properly.
+   - [Estimate](./estimation/readme.md) the effort for each backlog item, as a team.
```
[the remark this answers is not in the document]
Comment (meulta) [reply sprint-planning/readme.md]: These are fixed in a recent update. Using story everywhere now :)

File: sprint-planning/readme.md [hunk 9]
```diff
@@ -1,29 +1,72 @@
...
+**During the meeting:**
+- Confirming team capacity. This should be done using the average velocity (number of points achieved per sprint) and removing the project vacations/holidays/off days. If it helps, you can get the daily average velocity, divide by the number of team members and multiply by number of off days.
+- Timebox (for instance: half of the meeting for story selection, half of the meeting for task splitting).
+- Agree on how much capacity needs to be "saved" for bug fixing (might depend on the sprint).
+- Ensure everyone understands each story that is selected for the sprint.
+- Ensure everyone participates in story effort estimation.
+
+**Other considerations:**
+- Take into account off days (vacations, national holidays, unavailability).
+- If the amount of work in a Sprint is is too much to handle, you might want to split into different workstreams. This might require thinking about [scrum of scrums](./../scrum-of-scrums/readme.md) and all related ceremonies.
```
[the remark this answers is not in the document]
Comment (meulta) [reply sprint-planning/readme.md]: Fixed

File: sprint-planning/readme.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (meulta) [reply sprint-planning/readme.md:15]: Fixed

[the remark this answers is not in the document]
Comment (meulta) [reply sprint-planning/readme.md:15]: Fixed

File: sprint-planning/readme.md [hunk 10]
```diff
@@ -1,29 +1,72 @@
...
-- Complete outline below
-- Determining team capacity
-- Tasking sprint stories
-  - size of tasks
-- WIP limits / dividing up into workstreams
+During the [sprint planning](https://www.agilealliance.org/glossary/sprint-planning), the team discusses and agrees on the scope for the upcoming sprint. 
 
-## Goals
+**Goals are:**
+- Selecting the **stories** that will be implemented in the sprint.
```
[the remark this answers is not in the document]
Comment (meulta) [reply sprint-planning/readme.md:9]: Fixed to using story/stories everywhere

File: sprint-planning/readme.md [hunk 11]
```diff
@@ -1,29 +1,72 @@
...
+- Estimating the **effort** required for the stories in the sprint.
+- Splitting the stories into **tasks**.
```
[the remark this answers is not in the document]
Comment (meulta) [reply sprint-planning/readme.md:11]: Resolving this one as there is already an ongoing conversation about splitting into tasks

[the remark this answers is not in the document]
Comment (meulta) [reply sprint-planning/readme.md:11]: Already have a conversation going on in another comment

**Comment by meulta** [comment]
Made some updates, still have some 2 conversations opened. 
Let me know what you think is right.

File: sprint-planning/readme.md (hunk 3 above)
Comment (TessFerrandez) [reply sprint-planning/readme.md:46]: We have also split the stories into tasks during planning, partially to ensure everyone is on the same page and the acceptance criteria are understood and will be met

File: sprint-planning/readme.md [hunk 12]
```diff
@@ -1,29 +1,83 @@
...
 
-- Complete outline below
-- Determining team capacity
-- Tasking sprint stories
-  - size of tasks
-- WIP limits / dividing up into workstreams
+During the [sprint planning](https://www.agilealliance.org/glossary/sprint-planning), the team discusses and agrees on the scope for the upcoming sprint.
 
-## Goals
+**Goals are:**
```
Comment (TessFerrandez) [inline sprint-planning/readme.md:7]: Nit: Goals are: -> Goals:

Suggestion - can we avoid mixing **bold** and non bold list titles in the document, I think this would be as readable without the bold, but if you feel the bold is necessary I would suggest also making General guidance bold.

File: sprint-planning/readme.md [hunk 13]
```diff
@@ -1,29 +1,83 @@
...
+
+- Selecting the **stories** that will be implemented in the sprint.
```
Comment (TessFerrandez) [inline sprint-planning/readme.md:9]: Suggestion: Keep the same form on all the list items throughout the document.  I would propose imperative style - i.e. 
- Select the **stories** that will be implemented in the sprint
- Estimate the **effort** required for the stories in the sprint
- Split the stories into **tasks**

File: sprint-planning/readme.md [hunk 14]
```diff
@@ -1,29 +1,83 @@
...
 
 ## Participation
 
-What members of the team should participate?
+**Everyone** in the team should participate in the sprint planning.
+
+**Specific roles:**
+
+- **[Scrum Master](https://www.agilealliance.org/glossary/scrum-master)**:
+  - Facilitates the conversation.
```
Comment (TessFerrandez) [inline sprint-planning/readme.md:26]: See previous comment regarding imperative

- Facilitate the conversation
...

File: sprint-planning/readme.md [hunk 15]
```diff
@@ -1,29 +1,83 @@
...
+- How many stories needed more work before being presented (and were rejected because they were not clear enough)?
+- How realistic was the estimation of effort?
+- Team satisfaction (can be assessed during the retrospective): does everyone in the team feel included in sprint planning?
 
 ## Facilitation Guidance
 
-How might one go about running the ceremony? Is there a general pattern that the ceremony should follow?
+**Prior to the meeting:**
+
+- [ ] Set sprint goal.
```
Comment (TessFerrandez) [inline sprint-planning/readme.md:68]: These check boxes feel a bit misplaced here as they are not used in the rest of the document.  Without them it would follow the same form as the rest of the text

File: sprint-planning/readme.md [hunk 16]
```diff
@@ -1,29 +1,83 @@
...
+General guidance:
 
-What are the intended outcomes of the ceremonies?
+- The sprint planning should happen at the beginning of the new sprint (or at the end of the previous one). It usually lasts between 1 and 4 hours depending on the size of the team and duration of the sprint.
+- Each story should be able to be completed within the duration of the sprint. Otherwise, the story should be broken up into multiple stories.
+- Each task duration should be somewhere between 2 to 8 hours.
 
 ## Participation
 
-What members of the team should participate?
```
Comment (madossan01) [inline sprint-planning/readme.md:17]: As the PO is an important member in this meeting, I suggest to reinforce that PO participation is strongly recommended or even mandatory.

File: sprint-planning/readme.md [hunk 17]
```diff
@@ -1,29 +1,83 @@
...
+- **[Scrum Master](https://www.agilealliance.org/glossary/scrum-master)**:
+  - Facilitates the conversation.
+  - Ensures everyone is heard.
+  - Reminds scrums principles and sprint planning goals if necessary.
+- **[Product owner](https://www.agilealliance.org/glossary/product-owner/)**:
+  - Prior to the sprint planning: performs some [backlog grooming](../backlog-management/grooming/readme.md) to ensure that each story that they want to propose for the new sprint (*) :
+  
+    - Is in the correct position in the backlog, by right priority order.
+    - Contains a clear description.
+    - Contains a clear [definition of done](./../team-agreements/definition-of-done/readme.md).
```
Comment (madossan01) [inline sprint-planning/readme.md:34]: These two orientations seem to be a definition of ready content. So, I suggest to remove these two orientations and keep here the request to ensure that each story "is attending the definition of ready".

File: sprint-planning/readme.md [hunk 18]
```diff
@@ -1,29 +1,83 @@
...
+  - (Optional) self assign first task to team members.
+
+*(\*) some teams find useful to define a **[Definition of ready](./../team-agreements/definition-of-ready/readme.md)** that describes the list of things that needs to be done in each story before the **product owner** can propose it for a **sprint**. The list proposed here is the classic minimal definition of ready.*
 
 ## Impact
 
-What positive impact has been observed as a result of this practice?
+**Sprint planning key benefits:**
+
+- Everyone participates, the entire team is aware of the scope of the sprint.
```
Comment (madossan01) [inline sprint-planning/readme.md:54]: Besides this one, another important benefit is to have an agreement on the goal of the sprint and to make sure the team agrees on the stories that can be complete within the sprint.

File: sprint-planning/readme.md [hunk 19]
```diff
@@ -1,29 +1,83 @@
...
+- Each team member takes responsibility in the sprint scope by participating in the stories discovery, sign up, prioritization and estimation.
```
Comment (madossan01) [inline sprint-planning/readme.md:55]: I suggest to remove the sign up from this statement as it refers to the story. 
As already commented by Brian above, the story is owned by the team.

File: sprint-planning/readme.md [hunk 20]
```diff
@@ -1,29 +1,83 @@
...
+- Creates a platform to communicate, discover and discuss dependencies.
```
Comment (madossan01) [inline sprint-planning/readme.md:56]: Please replace platform by "channel" or "practice".

File: sprint-planning/readme.md [hunk 21]
```diff
@@ -1,29 +1,72 @@
...
+**Sprint planning key benefits:**
+- Everyone participates, the entire team is aware of the scope of the sprint.
+- Each team member takes responsibility in the sprint scope by participating in the stories discovery, sign up, prioritization and estimation.
+- Creates a platform to communicate, discover and discuss dependencies.
 
 ## Measures
 
-How might one measure the impact of this ceremony positive or negative?
+- How many stories needed more work before being presented (and were rejected because not clear enough)?
+- How realistic was the estimation of effort?
```
[the remark this answers is not in the document]
Comment (madossan01) [reply sprint-planning/readme.md:61]: An interesting measurement may be the  #of stories in the sprint that reach the Goal x #of stories in the sprint that does not reach the goal.

File: sprint-planning/readme.md (hunk 12 above)
Comment (meulta) [reply sprint-planning/readme.md:7]: Good suggestion. Removed it in upcoming update

File: sprint-planning/readme.md (hunk 13 above)
Comment (meulta) [reply sprint-planning/readme.md:9]: Changed in upcoming update

File: sprint-planning/readme.md (hunk 14 above)
Comment (meulta) [reply sprint-planning/readme.md:26]: Changed in upcoming update

File: sprint-planning/readme.md (hunk 15 above)
Comment (meulta) [reply sprint-planning/readme.md:68]: Agreed, removed in upcoming change

File: sprint-planning/readme.md (hunk 21 above)
[the remark this answers is not in the document]
Comment (madossan01) [reply sprint-planning/readme.md:62]: This measurement can be added in the next update.

File: sprint-planning/readme.md [hunk 22]
```diff
@@ -1,29 +1,72 @@
...
+     - Contains a clear description.
+     - Contains a clear [definition of done](https://www.agilealliance.org/glossary/definition-of-done/).
+     - Do NOT pre assigns stories to the future sprint. This is the purpose of the sprint planning.
+  - During the meeting: 
+    - Describes to the team the stories that they propose for the sprint.
+-  **All team members**:
+   - Listen to the product owner story description.
+   - Ask questions to make sure everyone understands each story properly.
+   - [Estimate](./estimation/readme.md) the effort for each backlog item, as a team. 
+   - Split each story into tasks.
```
[the remark this answers is not in the document]
Comment (sarath-p) [reply sprint-planning/readme.md]: The dev team usually breaks down the story into tasks in the planning process by taking design and technical constrains into consideration assuming the Product Owner writing the story. if the story is clear with no or less ambiguity and clear definition of done functionally, it should be very quick.

File: sprint-planning/readme.md [hunk 23]
```diff
@@ -1,29 +1,72 @@
...
+   - (Optional) self assign first task to team members.
```
Comment (sarath-p) [inline sprint-planning/readme.md]: In addition to optional assignment, another bullet point would be helpful to prioritize task list so that the team can pick up next available task based on the priority when they are done with existing task.

File: sprint-planning/readme.md [hunk 24]
```diff
@@ -1,29 +1,72 @@
...
+
+*(\*) some teams find useful to define a **[Definition of ready](https://www.agilealliance.org/glossary/definition-of-ready)** that describes the list of things that needs to be done in each story before the **product owner** can propose it for a **print**. The list proposed here is the classic minimal definition of ready.*
 
 ## Impact
 
-What positive impact has been observed as a result of this practice?
+**Sprint planning key benefits:**
+- Everyone participates, the entire team is aware of the scope of the sprint.
+- Each team member takes responsibility in the sprint scope by participating in the stories discovery, sign up, prioritization and estimation.
```
Comment (sarath-p) [inline sprint-planning/readme.md]: Is story discovery happens to move them to the sprint from backlog? Or new stories? If the new story is identified as part of the planning, it will be added to the backlog first and Product Owner prioritizes it by collaborating with dev/tech lead.

File: sprint-planning/readme.md (hunk 9 above)
Comment (sarath-p) [inline sprint-planning/readme.md]: I would suggest breaking stories into smaller stories to complete in the sprint based on teams velocity. The scrum of scrum is good when the team has more than 10-12 people so making team smaller improves efficiency in some areas.
